async_producer.go

```diff
@@ -1239,15 +1258,18 @@ func (p *asyncProducer) retryHandler() {
 		}
 
 		buf.Add(msg)
+		currentByteSize += int64(msg.ByteSize(version))
 
-		if maxBufferSize > 0 && buf.Length() >= maxBufferSize {
+		if (maxBufferLength > 0 && buf.Length() >= maxBufferLength) || (maxBufferBytes > 0 && currentByteSize >= maxBufferBytes) {
```

Comment:
[improvement?] Hm… 🤔  an idea would be to maybe invert the `if`, so the body could be an early `continue`. This follows https://go.dev/wiki/CodeReviewComments#indent-error-flow

Reply:
thanks. I've taken your advice and inverted the `if` condition via De Morgan's laws. This improvement also reduces the indentation level.